sdk/provisioner.go

```diff
@@ -102,8 +102,17 @@ func (out *ProvisionOutput) AddEnvVar(name string, value string) {
 	out.Environment[name] = value
 }
 
-// AddArgs can be used to add additional arguments to the command line of the provision output.
-func (out *ProvisionOutput) AddArgs(args ...string) {
+// AddArgs can be used to add additional arguments to the command line of the provision output, immediately after the executable name or at the end, depending on the value of provisionImmediatelyAfterExecutable.
+func (out *ProvisionOutput) AddArgs(provisionImmediatelyAfterExecutable bool, args ...string) {
```

Comment:
Two things I'd consider here:
- The change to this signature is breaking. This means that anyone in the process of developing a plugin will have to rewrite any usage of this. This is probably not too big of a deal, but something we should try to avoid as much as possible.
- A boolean as the argument makes it a bit difficult to see what a call to this function actually does. If you write `AddArgs(true, "argument")`, it's not clear to the reader what this `true` value means without looking at the function definition. If we instead use an enum for this, we can make it very easy to read. Consider something like this:

```go

type ArgumentPosition int

const (
       ArgumentPositionFirst ArgumentPosition = iota
       ArgumentPositionLast
)

output.AddArgs(ArgumentPositionFirst, "argument")
```
Possibly there are some better variants than "First" and "Last", but the idea here is that forcing the caller to use a constant, makes the code more readable.